tests/utils/datavolume.go

```diff
@@ -873,25 +873,27 @@ func WaitForConditions(ci ClientsIface, dataVolumeName, namespace string, timeou
 	}, timeout, pollingInterval).Should(gomega.BeTrue())
 }
 
-// verifyConditions checks if the conditions match testConditions
+// verifyConditions checks if all the conditions exist and match testConditions
 func verifyConditions(actualConditions []cdiv1.DataVolumeCondition, testConditions []*cdiv1.DataVolumeCondition) bool {
 	for _, condition := range testConditions {
-		if condition != nil {
-			actualCondition := dvc.FindConditionByType(condition.Type, actualConditions)
-			if actualCondition != nil {
-				if actualCondition.Status != condition.Status {
-					fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: Condition.Status does not match for type: %s, status expected: [%s], status found: [%s]\n", condition.Type, condition.Status, actualCondition.Status)
-					return false
-				}
-				if strings.Compare(actualCondition.Reason, condition.Reason) != 0 {
-					fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: Condition.Reason does not match for type: %s, reason expected [%s], reason found: [%s]\n", condition.Type, condition.Reason, actualCondition.Reason)
-					return false
-				}
-				if !strings.Contains(actualCondition.Message, condition.Message) {
-					fmt.Fprintf(ginkgo.GinkgoWriter, "INFO: Condition.Message does not match for type: %s, message expected: [%s],  message found: [%s]\n", condition.Type, condition.Message, actualCondition.Message)
-					return false
-				}
-			}
+		if condition == nil {
+			continue
+		}
+		actualCondition := dvc.FindConditionByType(condition.Type, actualConditions)
+		if actualCondition == nil {
```

Comment:
So the test was passing because at some point `actualCondition==nil` was true for all 3 of them?

Reply:
Even worse - it passed whenever the DV **existing** conditions matched the passed `testConditions`, so if you have several `testConditions` and none of them exists in the DV, the test would pass, and we were actully testing nothing here...

Reply:
Yeah I think that is what I meant `actualCondition==nil` means the DV had none of them at one point